Sources/JSONUtilities/Dictionary+JSONKey.swift

```diff
@@ -318,15 +256,48 @@ extension Dictionary where Key: StringProtocol {
     return jsonArray
   }
 
-  // MARK: JSONArray decoding
+  // MARK: Dictionary decoding
 
-  fileprivate func decodableObjectsArray<ReturnType: JSONObjectConvertible>(_ jsonArray: JSONArray) -> [ReturnType] {
-    return jsonArray.flatMap {
-      guard let castedJsonObject = $0 as? JSONDictionary else {
-        return nil
+  fileprivate func decodeDictionary<T>(atKeyPath keyPath: Key, invalidItemBehaviour: InvalidItemBehaviour = .remove, decodeClosure: (JSONDictionary, String) throws -> T?) throws -> [String: T] {
+    let jsonDictionary: JSONDictionary = try json(atKeyPath: keyPath)
+
+    var dictionary: [String: T] = [:]
+    for (key, _) in jsonDictionary {
+
+      switch invalidItemBehaviour {
+      case .remove:
+        if let typedItem = try? decodeClosure(jsonDictionary, key) {
```

Comment:
Is worth flipping the order of the `jsonDictionary, key` to `key, jsonDictionary` as this matches the `[String: T]`?